### PR TITLE
Feature/raw packet features

### PR DESCRIPTION
**Pull request by RadionBik**
Hi!

This is a reimplementation of raw packet-features plugin, that I am using in my project. i thought it would be useful to have such a usage example, and it might be purely optional component, that one can use importing from the `plugin` module. Please, take a look and tell me what you think about it.

### CODE REVIEW

**Comment by aouinizied** [comment]
@RadionBik Thanks for contributing. Yes, I believe that the main goal of NFStream project is the plugins part. Consequently, researchers may share their plugins and this would help to improve reproducibility  of work in this field.

Additionally, my future plans are to add method that allow selection of Plugins that will run in NFSTream. Example: NFStreamer(metering_arguments)["flow_feat_1", "flow_feat_2"] will run NFSTream computing only this two features. We will need to have a kind of dependency graph between NFPLugins but it's possible.

Here my remarks regarding your PR before merging:
*  We could change __init__ prototype of NFPlugin to be more pythonic. Removing the user_data stuff and you will have all your additional arguments with *args and  **kwargs. What do you think about.
* You are returning a value at your on_update method which is not used. Could you remove the return part as it may be confusing for users that will try to build plugins based on your example.

Zied

File: nfstream/plugin.py [hunk 1]
```diff
@@ -1291,6 +1294,119 @@ def on_update(self, obs, entry):
...
+                raw_feature_matrix[counter, self._active_builtin_features_count + idx] = handler(obs)
+        return raw_feature_matrix
+
+    def on_init(self, obs):
+        raw_feature_matrix = np.zeros((self.packet_limit, self._active_feature_count))
+        return self._fill_flow_stats(obs, raw_feature_matrix)
+
+    def on_update(self, obs, entry):
+        if entry.bidirectional_packets > self.packet_limit:
+            return entry.raw_packets_matrix
```
Comment (aouinizied) [inline nfstream/plugin.py:1402]: pass

File: nfstream/plugin.py [hunk 2]
```diff
@@ -1291,6 +1294,119 @@ def on_update(self, obs, entry):
...
+        return self._fill_flow_stats(obs, entry.raw_packets_matrix, counter=entry.bidirectional_packets - 1)
```
Comment (aouinizied) [inline nfstream/plugin.py:1403]: without the return

**Comment by RadionBik** [comment]
I addressed the remarks you provided, please, take a look. Also, I would like to mention that setting up the dev environment could be easier, if there were building instructions for the .so lib. I am not sure about the best-practices for that, but I had to move `setup_observer_cc()` and `setup_ndpi()` functions to a separate callable module to accomplish that task. Is there the right way to do that?

**Comment by aouinizied** [comment]
@RadionBik which command did you use to setup your build environnent?

**Comment by RadionBik** [comment]
I didn't used any specific command. As I said, I copied the functions to a python script and executed it, so as to build .so without installing. Was there a command? 😅

**Comment by aouinizied** [comment]
Once you are in the cloned nfstream repo:
```bash
sudo apt-get update
sudo apt-get install autoconf automake libtool pkg-config libpcap-dev libzmq3-dev
sudo apt-get install libusb-1.0-0-dev libdbus-glib-1-dev libbluetooth-dev libnl-genl-3-dev flex bison
pip install -r requirements.txt
python setup.py bdist_wheel
```
Whatever you want to test, the script must be under the root of repository and that's it.
If the native part is modified, you need to rerun (python setup.py bdist_wheel)

**Comment by RadionBik** [comment]
I just the skipped the last command, I thought it would install the package, and now I realized that it builds binaries. Thank you!

**Pull request merged**